design/ceph-osd-removal.md

```diff
@@ -0,0 +1,113 @@
+# Removing OSDs from a Ceph cluster
+
+**Targeted for v1.X**
+
+## Background
+
+Removing OSDs is a common day-2 operation and users need to be able to perform
+this regularly.
+
+With the effort to add OSD replacement in the Ceph Dashboard, there is a demand
+for automating the deletion of individual OSDs.
+
+## Status
+
+As of Rook 1.1, there is some preliminary support for removing OSDs:
+
+### Removing Nodes
+
+Rook fully supports the removal of whole nodes from a Cluster, including all
+OSDs. This implementation does every call needed to completely remove all OSDs
+on that specific node from a Ceph cluster if that cluster no longer appears in
+the list of nodes in the CephCluster CR.
+
+There are concerns that this implementation might be dangerous, if a node is
+being removed inadvertently.
```

Comment:
How about we state that we're removing that functionality? I think we already agree on that point.

Reply:
done